src/Atmos/Model/reconstructions.jl

```diff
@@ -0,0 +1,89 @@
+using KernelAbstractions.Extras: @unroll
+
+export HBFVReconstruction
+
+using ..DGMethods.FVReconstructions: AbstractReconstruction
+import ..DGMethods.FVReconstructions: width
+
+struct HBFVReconstruction{M} <: AbstractReconstruction
+    _atmo::M
+    _recon::AbstractReconstruction
+end
+
+width(hb_recon::HBFVReconstruction) = width(hb_recon._recon)
+
+function (hb_recon::HBFVReconstruction)(
+    state_bot,
+    state_top,
+    cell_states::SVector{D},
+    cell_weights,
+) where {D}
+
+
+    FT = eltype(state_bot)
+    vars_prim = Vars{vars_state(hb_recon._atmo, Primitive(), FT)}
+    _grav = FT(grav(hb_recon._atmo.param_set))
```

Comment:
Would it be possible to define `_grav` as `vert_unit_vector \cdot gradient(geopotential)`? The advantage of doing that is that if we want to generalize the model to deep atmospheres, for which `g` is not constant, it goes through straightforwardly. That being said, if it's much easier to use fixed `g` now, that's fine too. We will not need such deep atmospheres anytime soon (we'll need when we want to simulate Jupiter or Saturn, which would be interesting one day...)

Reply:
I will start an issue to document the non constant ```g``` case, maybe update the this in another PR, since we assume the ```g``` is constant in many places
- reference profile
- discrete reference/initialization condition
- geopotential computation
- hydrostatic balanced reconstruction